agent/structs/config_entry.go

```diff
@@ -613,6 +624,9 @@ type ServiceConfigRequest struct {
 	// Mode indicates how the requesting proxy's listeners are dialed
 	Mode ProxyMode
 
+	// NoUpstreams forces just data about the specified service to be returned and no related data.
+	NoUpstreams bool
```

Comment:
This bypasses all upstream related data collection, even in tproxy mode to save on resources.